drivers/dai/intel/dmic/dmic.c

```diff
@@ -841,6 +841,16 @@ static int dai_dmic_remove_wrapper(const struct device *dev)
 	return ret;
 }
 
+static inline void dai_dmic_irq_init(void)
+{
+	IRQ_CONNECT(
+		DT_INST_IRQN(0),
+		IRQ_DEFAULT_PRIORITY,
+		dai_dmic_irq_handler,
+		DEVICE_DT_INST_GET(0),
+		0);
+}
```

Comment:
This is not making too much sense to me. That is a static configuration, why don't you do it in `dai_dmic_initialize_device` ?